app/views/handlers/individual_response.py

```diff
@@ -130,6 +139,58 @@ def form(self):
     def get_context(self):
         return build_question_context(self.rendered_block, self.form)
 
+    def _get_fulfilment_code(self, fulfilment_type):
+        fulfilment_codes = {
+            "sms": {
+                GB_ENG_REGION_CODE: "UACITA1",
+                GB_WLS_REGION_CODE: "UACITA2B",
+                GB_NIR_REGION_CODE: "UACITA4",
+            },
+            "postal": {
+                GB_ENG_REGION_CODE: "P_UAC_UACIP1",
+                GB_WLS_REGION_CODE: "P_UAC_UACIP2B",
+                GB_NIR_REGION_CODE: "P_UAC_UACIP4",
+            },
+        }
+
+        region_code = self._metadata["region_code"]
+        return fulfilment_codes[fulfilment_type][region_code]
+
+    def _get_fulfilment_request_payload(self, mobile_number=None):
+        fulfilment_type = "sms" if mobile_number else "postal"
+        individual_case_id_mapping = (
+            {}
+            if self._metadata.get("case_type") == "SPG"
+            else {"individualCaseId": str(uuid4())}
+        )
+
+        mobile_number_mapping = {"telNo": mobile_number} if mobile_number else {}
+
+        message = {
+            "event": {
+                "type": "FULFILMENT_REQUESTED",
+                "source": "QUESTIONNAIRE_RUNNER",
+                "channel": "EQ",
+                "dateTime": datetime.now(tz=tzutc()).isoformat(),
+                "transactionId": str(uuid4()),
+            },
+            "payload": {
+                "fulfilmentRequest": {
+                    **individual_case_id_mapping,
+                    "fulfilmentCode": self._get_fulfilment_code(fulfilment_type),
+                    "caseId": self._metadata["case_id"],
+                    "contact": mobile_number_mapping,
+                }
+            },
+        }
+
+        return json.dumps(message).encode("utf-8")
```

Comment:
As discussed, the building of the payload should be in it's own class, rather than part of the handler.

Reply:
Added a new class to fetch the payload.